R/otp-isochrone.R

```diff
@@ -0,0 +1,111 @@
+#' Get the Isochrones from a location
+#'
+#' @param otpcon OTP connection object produced by otp_connect()
+#' @param fromPlace Numeric vector, Latitude/Longitude pair, e.g. `c(51.529258,-0.134649)`
+#' @param mode Character vector of modes of travel valid values TRANSIT, WALK, BICYCLE, CAR, BUS, RAIL, default CAR
+#' @param date_time POSIXct, a date and time, defaults to current date and time
+#' @param arriveBy Logical, Whether the trip should depart or arrive at the specified date and time, default FALSE
+#' @param maxWalkDistance Numeric passed to OTP
+#' @param walkReluctance Numeric passed to OTP
+#' @param transferPenalty Numeric passed to OTP
+#' @param minTransferTime Numeric passed to OTP
+#' @param cutoffSec Numeric vector, number of seconds to define the break points of each Isochrone
+#' @return
+#' Returns a data.frame of SF POLYGONS
+#' @examples
+#' isochrone1 <- otp_isochrone(otpcon, fromPlace = c(51.5292,-0.1346))
+#' isochrone2 <- otp_isochrone(otpcon, fromPlace = c(51.5292,-0.1346), mode = c("WALK","TRANSIT"), cutoffSec = c(600,1200,1800))
+#' @detials Isochrones are maps of equal travel time,
+#' for a given location a map is produced shoing how long it takes to reach
+#' each location.
+#' @export
+otp_isochrone <- function(otpcon = NA,
+                     fromPlace = NA,
+                     mode = "CAR",
+                     date_time = Sys.time(),
+                     arriveBy = FALSE,
+                     maxWalkDistance = 800,
+                     walkReluctance = 5,
+                     transferPenalty = 0,
+                     minTransferTime = 600,
+                     cutoffSec = c(600,1200,1800,2400, 3000, 3600))
```

Comment:
@mem48 I don't think CAR should be the default here as it isn't a valid mode for isochrones. It will return polygons but they won't be correct. Need to think about whether we even allow non-transit modes as options here.